src/core/schema.js

```diff
@@ -138,6 +138,7 @@ function parseProperty (value, propDefinition) {
   // Use default value if value is falsy.
   if (value === undefined || value === null || value === '') {
     value = propDefinition.default;
+    if (Array.isArray(value)) { value = value.slice(); }
```

Comment:
To describe the situation that could cause unexpectedness (similar to when I found out components were messing with each other's arrays):

```
AFRAME.registerComponent('foo', {
  schema: {
    someData: { 
      default: {value: 0}, parse: function () { // ... }
    }
  }

  init: function () {
    this.data.someData.value += 1;
  }
});

<a-entity id="a" foo></a-entity>
<a-entity id="b" foo>/a-entity>
```

Both instances of `foo` are sharing the same object and polluting each other's data sets. Entity A and Entity B would be both surprised that somehow their values are now `2` when I'd expect `1`.

This is also inconsistent that if `foo` did define data, it would no longer share the object. Only if they use the default value, it'd be copied once per initialization (in most of my cases in creating schemaless components, it'd be copying an empty object).